1436790/1438462: Part 17: 19 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
Commit the second occurrence of a topic fix mentioned in part 14 (PR#10374).

DennisLee-DennisLee committed:
Merge branch 'master' of https://github.com/dotnet/docs into v-dele-1438462-017

DennisLee-DennisLee committed:
19 files with MSDN link fixes.

DennisLee-DennisLee committed:
Fixed the merge conflict.

DennisLee-DennisLee committed:
Fixed some previous-version links to go to the dotnet/wcf-data-services repo instead of visualstudio (to show a richer TOC).

DennisLee-DennisLee committed:
Fixed a straggler.

mairaw commented on docs/framework/debug-trace-profile/enabling-jit-attach-debugging.md:
```diff
@@ -32,4 +32,4 @@ JIT-attach debugging is the phrase used to describe attaching a debugger to a pr
 - [Debugging, Tracing, and Profiling](../../../docs/framework/debug-trace-profile/index.md)
 - [Making an Image Easier to Debug](../../../docs/framework/debug-trace-profile/making-an-image-easier-to-debug.md)
-- [Enabling Profiling](https://msdn.microsoft.com/library/3b669676-f0e0-4ebf-8674-68986dd2020d)
+- [Enabling Profiling](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/s5ec0es1(v=vs.100))
```
```suggestion
```

mairaw commented on docs/framework/development-guide.md:
```diff
@@ -73,5 +73,5 @@ This section explains how to create, configure, debug, secure, and deploy your .
  Describes the tools that help you develop, configure, and deploy apps by using .NET Framework technologies.  
   
- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)  
+ [.NET Framework Samples](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/bb400848(v=vs.100))  
```
```suggestion
 [.NET Samples](../samples-and-tutorials/index.md)  
```

Please change the description below as well

mairaw commented on docs/framework/get-started/overview.md:
```diff
@@ -83,7 +83,7 @@ As you would expect from an object-oriented class library, the .NET Framework ty
 - Service-oriented apps using Windows Communication Foundation (WCF). See [Service-Oriented Applications with WCF](../../../docs/framework/wcf/index.md).
 
-- Workflow-enabled apps using Windows Workflow Foundation (WF). See [Building Workflows in the .NET Framework](https://msdn.microsoft.com/library/cbf3880f-dc7b-466d-b808-1109b1223f4a).
+- Workflow-enabled apps using Windows Workflow Foundation (WF). See [Windows Workflow Foundation](/docs/framework/windows-workflow-foundation/index.md).
```
```suggestion
- Workflow-enabled apps using Windows Workflow Foundation (WF). See [Windows Workflow Foundation](../windows-workflow-foundation/index.md).
```

mairaw commented on docs/framework/get-started/overview.md:
```diff
@@ -93,5 +93,5 @@ The Windows Forms classes are a comprehensive set of reusable types that vastly
 - [Development Guide](../../../docs/framework/development-guide.md)
 - [Tools](../../../docs/framework/tools/index.md)
-- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)
+- [.NET Framework Samples](/docs/samples-and-tutorials/index.md)
```
```suggestion
- [.NET Framework Samples](../../samples-and-tutorials/index.md)
```

DennisLee-DennisLee replied:
Just curious - why don't you allow in-repo links that begin with a slash and end with *ArticleName*.md? The Contributor Guide recommends that to avoid links that begin with a tilde. It seems like a good way to avoid copy-and-paste problems for common links that need to be fixed in different folders/folder levels (like the problem with parent folder counting in one of my previous PRs).

https://github.com/Microsoft/vscode-docs-authoring/issues/135

mairaw commented on docs/framework/get-started/system-requirements.md:
```diff
@@ -92,7 +92,7 @@ The .NET Framework requires administrator privileges for installation. If you do
 - [!INCLUDE[winserver8](../../../includes/winserver8-md.md)] includes the [!INCLUDE[net_v45](../../../includes/net-v45-md.md)], so you don't have to install it separately. Similarly, [!INCLUDE[winblue_server_2](../../../includes/winblue-server-2-md.md)] includes the [!INCLUDE[net_v451](../../../includes/net-v451-md.md)].
 
-- The .NET Framework has limited support for the Server Core Role with Windows Server 2008 R2 SP1 or later. See [Server Core .NET Functionality](https://msdn.microsoft.com/library/ee391632.aspx) for a list of unsupported APIs.
+- The .NET Framework has limited support for the Server Core Role with Windows Server 2008 R2 SP1 or later. See [Server Core .NET Functionality](https://docs.microsoft.com/previous-versions//dd745015(v=vs.85)) for a list of unsupported APIs.
```
@rpetrusha would it be okay to remove this sentence now?

mairaw commented on docs/framework/interop/blittable-and-non-blittable-types.md:
```diff
@@ -43,15 +43,15 @@ Most data types have a common representation in both managed and unmanaged memor
 -   One-dimensional arrays of blittable types, such as an array of integers. However, a type that contains a variable array of blittable types is not itself blittable.  
   
--   Formatted value types that contain only blittable types (and classes if they are marshaled as formatted types). For more information about formatted value types, see [Default Marshaling for Value Types](https://msdn.microsoft.com/library/4d9a876c-e05a-40ba-bd85-bd22877f984a(v=vs.100)).  
+-   Formatted value types that contain only blittable types (and classes if they are marshaled as formatted types). For more information about formatted value types, see [Default Marshaling for Value Types](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/0t2cwe11(v=vs.100)).  
```
```suggestion
-   Formatted value types that contain only blittable types (and classes if they are marshaled as formatted types). For more information about formatted value types, see [Default marshaling for value types](default-marshaling-behavior.md#default-marshaling-for-value-types).  
```

mairaw commented on docs/framework/interop/blittable-and-non-blittable-types.md:
```diff
@@ -43,15 +43,15 @@ Most data types have a common representation in both managed and unmanaged memor
  Some managed data types require a different representation in an unmanaged environment. These non-blittable data types must be converted into a form that can be marshaled. For example, managed strings are non-blittable types because they must be converted into string objects before they can be marshaled.  
   
- The following table lists non-blittable types from the <xref:System> namespace. [Delegates](https://msdn.microsoft.com/library/d176ee76-f982-494b-b03d-92e4118896e2(v=vs.100)), which are data structures that refer to a static method or to a class instance, are also non-blittable.  
+ The following table lists non-blittable types from the <xref:System> namespace. [Delegates](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/7esfatk4(v=vs.100)), which are data structures that refer to a static method or to a class instance, are also non-blittable.  
```
```suggestion
 The following table lists non-blittable types from the <xref:System> namespace. [Delegates](default-marshaling-behavior.md#default-marshaling-for-delegates), which are data structures that refer to a static method or to a class instance, are also non-blittable.  
```

mairaw reviewed (changes requested):
Thanks @DennisLee-DennisLee. Some comments as usual 😺

mairaw committed:
Apply suggestions from code review

Accepting all eight of Maira's suggestions.

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

mairaw commented on docs/framework/development-guide.md:
```diff
@@ -73,5 +73,5 @@ This section explains how to create, configure, debug, secure, and deploy your .
  Describes the tools that help you develop, configure, and deploy apps by using .NET Framework technologies.  
   
- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)  
+ [.NET Samples](../samples-and-tutorials/index.md)  
```
```suggestion
 [.NET samples and tutorials](../samples-and-tutorials/index.md)  
```

mairaw commented on docs/framework/development-guide.md:
```diff
@@ -73,5 +73,5 @@ This section explains how to create, configure, debug, secure, and deploy your .
   
- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)  
+ [.NET Samples](../samples-and-tutorials/index.md)  
  Provides links to the MSDN Code Samples Gallery for sample apps that demonstrate .NET Framework technologies.
```
```suggestion
 Provides links to samples and tutorials that help you learn about .NET.
```

mairaw commented on docs/framework/get-started/overview.md:
```diff
@@ -93,5 +93,5 @@ The Windows Forms classes are a comprehensive set of reusable types that vastly
 - [Development Guide](../../../docs/framework/development-guide.md)
 - [Tools](../../../docs/framework/tools/index.md)
-- [.NET Framework Samples](https://msdn.microsoft.com/library/177055f8-4a1f-43e7-aee6-995c196079b1)
+- [.NET Framework Samples](../../samples-and-tutorials/index.md)
```
```suggestion
- [.NET samples and tutorials](../../samples-and-tutorials/index.md)
```

mairaw reviewed (changes requested):
A few things still left

mairaw committed:
Apply suggestions from code review

Committing 3 additional suggestions from Maira (involving .NET Framework samples and tutorials).

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>

mairaw reviewed (approved):
Thanks @DennisLee-DennisLee!
